airflow/models.py

```diff
@@ -1414,6 +1411,15 @@ def get_template_context(self, session=None):
             'test_mode': self.test_mode,
         }
 
+        # Allow task level param definitions to be rendered via jinja
+        # using the context available up until this point
+        if task.params:
```

Comment:
This should be done in the `render_templates` and apply only on strings (use `isinstance(obj, basestring)` to check)